Automattic/sensei
# Add event logging for module creation with base properties

Closes #2648 
Closes #2652 

This PR adds logging for the module creation event, and the base properties to be sent with all event logging.

I'm using our usage tracking code to send the events, and there is one potential issue that I came across that I wanted to bring up. Our internal guidelines for creating event names says the following:

> Don’t programmatically create event names. They should be hardcoded so they are easily searchable in the code base.

We do create the event names programmatically because our code prepends the `sensei_` prefix [here](https://github.com/Automattic/sensei/blob/0212da5f1c816da9d911ab227e1f4e5555adbf40/includes/lib/usage-tracking/class-usage-tracking-base.php#L214).

I think this is fine in our case, since all of our events will have the same prefix. But if we think that this will cause confusion when searching the code for event names, we can change this.

## Testing instructions

- Go to the Courses > Modules page.
- Create a new module.
- Ensure a `sensei_module_add` event was logged.
- In the logged event, ensure it has the following properties:
  - `paid`
  - `courses`
  - `learners`
  - `page` (ensure that the value is `module`)
- See #2648 for the expected values of `paid` (should always be `0`, see https://github.com/Automattic/sensei/pull/2672#issuecomment-489172505), `courses`, and `learners`. Ensure that they are correct.
- Create another module, and select a parent for this one.
- Ensure the new logged event contains the property `parent_id` with the parent module's ID.
- Go to the Edit page for a course.
- From the metabox in the right sidebar, add a new module.
- Ensure that an event is logged, the `parent_id` does not exist, and the `page` is `course`.
- Repeat the above two instructions on the Add Course page.

## New hooks

- `sensei_event_logging_base_fields`: `includes/class-sensei-usage-tracking-data.php` line 80
